environs/config/config.go

```diff
@@ -621,6 +636,26 @@ func Validate(cfg, old *Config) error {
 		}
 	}
 
+	if v, ok := cfg.defined[RsyslogURL].(string); ok {
+		_, err := url.Parse(v)
+		if err != nil {
+			return fmt.Errorf("invalid rsyslog URL: %v", err)
+		}
+	}
+	rsyslogCACert, rsyslogCACertOK := cfg.RsyslogCACert()
```

Comment:
I feel like I was looking at very similar code somewhere else... which I think forgot to parse the url. I think there's a strong case for a `core/rsyslog.ClientConfig` with a `Validate()` method (and I guess an `IsZero()`? or a specific `rsyslog.ErrNoClientConfig`..? which I'm getting indications will become convenient before long)

and if we did that, we could plausibly just spam all the values into one of those structs for validation; *and* just return one from a significantly-narrowed `RsyslogClientConfig() (rsyslog.ClientConfig, error)` method on `config.Config`. Sane?